linera-views/src/common.rs

```diff
@@ -122,6 +125,121 @@ where
     }
 }
 
+/// GreatestLowerBoundIterator iterates over the entries of a BTreeSet.
```

Comment:
Actually, if I'm reading this correctly we're only using this to check whether a key starts with any of the deleted prefixes. Wouldn't the right data structure for that be a trie? In general, should we simply always store the set of deleted prefixes in the form of a trie? Then we wouldn't need this iteration, we wouldn't rely on the ordering, and looking up if something starts with a deleted prefix would be a natural operation, with a complexity only linear in the length of the looked-up key.

Reply:
I haven't tried the existing implementations, but it might sense to implement it ourselves anyway, because we want a few specific behaviors:
* Inserting a prefix of an existing entry should remove the existing entry.
* Inserting an extension of an existing entry should be a no-op.

Reply:
Ok, I am open to that. But I think we could do that in a subsequent work.